lib/src/chewie_controller.dart

```diff
@@ -0,0 +1,182 @@
+import 'package:chewie/src/chewie_progress_colors.dart';
+import 'package:flutter/material.dart';
+import 'package:video_player/video_player.dart';
+
+/// The state of the [ChewieController].
+@immutable
+class ChewieValue {
+  ChewieValue(
+    this.videoPlayerController, {
+    this.isFullScreen = false,
+  });
+
+  /// True if the video is currently playing fullscreen
+  final bool isFullScreen;
+
+  /// The controller for the video you want to play
+  final VideoPlayerController videoPlayerController;
+
+  ChewieValue copyWith({
+    VideoPlayerController videoPlayerController,
+    bool isFullScreen,
+  }) {
+    return ChewieValue(
+      videoPlayerController ?? this.videoPlayerController,
+      isFullScreen: isFullScreen ?? this.isFullScreen,
+    );
+  }
+
+  @override
+  String toString() {
+    return '$runtimeType('
+        'isFullscreen: $isFullScreen, '
+        'videoPlayerController: $videoPlayerController, ';
+  }
+}
+
+class ChewieController extends ValueNotifier<ChewieValue> {
```

Comment:
Could `ChewieController` be a `ChangeNotifier` instead of a `ValueNotifier<ChewieValue>`, and move the `isFullScreen` and `videoPlayerController` as part of it?

That way folks could just do `chewieController.isFullscreen` or `chewController.addListener`, etc. 

That way you could just access those

Reply:
It was modelled after the `VideoPlayerController` which extends `ValueNotifier<VideoPlayerValue>`. It has getters like the `controller.position` but other values are accessed like `controller.value.isPlaying` or `controller.value.volume`. But you are right it would be a nicer API if it can all be accessed as short as possible. I will look into using `ChangeNotifier` directly.

Reply:
Ah, you're right... If you have the ability, maybe take 5-10 minutes to quickly refactor it and see how it works out! 

If the current API you've got here seems fine, no worries at all.

Reply:
I could also add getters to the current `ChewieController`:
```dart
bool get isFullScreen => value.isFullScreen;
```
Will try to find out why some controllers out there are using `ValueNotifier` and others `ChangeNotifier`.

Reply:
Added the two getters. So the API is nice and short now. If at some point we should see big advantages in using the `ChangeNotifier` directly we can switch without breaking stuff for users.

Reply:
Switched to using a `ChangeNotifier`. We can switch back to using ValueNotifier when we have a lot more values and it is more convenient to change and notify with one `value.copyWith(foo='bar')` instead of explicitly having to call notifyListeners(). Until then the code is just much cleaner without so much boiler plate.